app/models/core/IncomeSourceId.scala

```diff
@@ -31,6 +32,17 @@ object IncomeSourceId {
 
   }
 
+  def compare(id: Option[IncomeSourceIdHash])(implicit user: MtdItUser[_]): Option[IncomeSourceId] = {
```

Comment:
we don't need to add this complexity to the type level
we should keep time as simple as possible